Docs for extended support for Kubernetes versions in EKS-A (#9231)
aws/eks-anywhere: docs/content/en/docs/clustermgmt/support/cluster-license.md

aliases:
    /docs/tasks/cluster/cluster-license/
description: >

>   Apply EKS Anywhere Enterprise Subscription licenses to your EKS Anywhere cluster

Reviewer: ```suggestion
  Apply EKS Anywhere Enterprise Subscription licenses to your EKS Anywhere clusters
```

---

When you purchase an EKS Anywhere Enterprise Subscription, licenses created in the AWS account and region you used to purchase the subscription. After purchasing your subscription, you can view your licenses, accept the license grants, and apply the license IDs or license tokens to your EKS Anywhere clusters. 

## Get license ID string or license token

> The two key parts of the license are the license ID string and the license token. In EKS Anywhere versions v0.21 and below, the license ID string is applied as a Kubernetes Secret to EKS Anywhere clusters and is used for AWS Support cases to validate the cluster is eligible for support. The license token was introduced in EKS Anywhere version v0.22 and all existing EKS Anywhere subscriptions have been updated with a license token for each license. The license token is applied in the EKS Anywhere cluster specification.

Reviewer: ```suggestion
The two key parts of the license are the license ID string and the license token. In EKS Anywhere versions v0.21.x and below, the license ID string is applied as a Kubernetes Secret to EKS Anywhere clusters and is used for AWS Support cases to validate the cluster is eligible for support. The license token was introduced in EKS Anywhere version v0.22.0 and all existing EKS Anywhere subscriptions have been updated with a license token for each license. The license token is applied in the EKS Anywhere cluster specification.
```

> You can use either the license ID string or the license token when you create AWS Support cases for your EKS Anywhere clusters. To use extended support for Kubernetes versions in EKS Anywhere, available for EKS Anywhere version v0.22 and above, your clusters must have a valid and active license token to be able to create and upgrade clusters using the Kubernetes extended support versions.

Reviewer: ```suggestion
You can use either the license ID string or the license token when you create AWS Support cases for your EKS Anywhere clusters. To use extended support for Kubernetes versions in EKS Anywhere, available for EKS Anywhere versions v0.22.0 and above, your clusters must have a valid and unexpired license token to be able to create and upgrade clusters using the Kubernetes extended support versions.
```

#### AWS Management Console

You can view the licenses for your subscription in the [EKS Anywhere section of the EKS console](https://console.aws.amazon.com/eks/home#/eks-anywhere) by clicking on the Name of your active subscription. The licenses panel is shown on the Subscription details page and contains the license ID string and the license token for each license associated with your subscription.

> If you are applying a license to an EKS Anywhere cluster using EKS Anywhere version v0.22 or above, copy the **license token** and proceed to [Apply license to EKS Anywhere cluster.]({{< relref "cluster-license#apply-license-to-eks-anywhere-cluster">}})

Reviewer: ```suggestion
If you are applying a license to an EKS Anywhere cluster using EKS Anywhere version v0.22.0 or above, copy the **license token** and proceed to [Apply license to EKS Anywhere cluster.]({{< relref "cluster-license#apply-license-to-eks-anywhere-cluster">}})
```

> 

Reviewer: One suggestion here would be to include a screenshot of the console page displaying the licenses section with the ID and the token.

Author: Yea I wanted to do this but I don't have access to the console changes yet

> If you are applying a license to an EKS Anywhere cluster using EKS Anywhere version v0.21 or below, copy the **license ID string** and proceed to [Apply license to EKS Anywhere cluster.]({{< relref "cluster-license#apply-license-to-eks-anywhere-cluster">}})

Reviewer: ```suggestion
If you are applying a license to an EKS Anywhere cluster using EKS Anywhere version v0.21.x or below, copy the **license ID string** and proceed to [Apply license to EKS Anywhere cluster.]({{< relref "cluster-license#apply-license-to-eks-anywhere-cluster">}})
```

#### AWS CLI

Use the following command to get the license ID strings and license tokens for each license associated with a subscription. Note, the command must be run with the same account that created the subscription.

- Replace `region-code` with the AWS Region that hosts your subscription (for example `us-west-2`).
- Replace `my-subscription` in the `--query` string with the `name` for your subscription.

```bash
aws eks list-eks-anywhere-subscriptions \
  --region 'region-code' \
  --query 'subscriptions[?name==`my-subscription`].licenses[]'
```

> If you are applying a license to an EKS Anywhere cluster using EKS Anywhere version v0.22 or above, copy the **license token** and proceed to [Apply license to EKS Anywhere cluster.]({{< relref "cluster-license#apply-license-to-eks-anywhere-cluster">}}) An example of the license token in the response is shown below.

Reviewer: ```suggestion
If you are applying a license to an EKS Anywhere cluster using EKS Anywhere version v0.22.0 or above, copy the **license token** and proceed to [Apply license to EKS Anywhere cluster.]({{< relref "cluster-license#apply-license-to-eks-anywhere-cluster">}}) An example of the license token in the response is shown below.
```

```
TODO
```

> If you are applying a license to an EKS Anywhere cluster using EKS Anywhere version v0.21 or below, copy the **license ID string** and proceed to [Apply license to EKS Anywhere cluster.]({{< relref "cluster-license#apply-license-to-eks-anywhere-cluster">}}) An example of the license ID string in the response is shown below.

Reviewer: ```suggestion
If you are applying a license to an EKS Anywhere cluster using EKS Anywhere version v0.21.x or below, copy the **license ID string** and proceed to [Apply license to EKS Anywhere cluster.]({{< relref "cluster-license#apply-license-to-eks-anywhere-cluster">}}) An example of the license ID string in the response is shown below.
```

```
TODO
```

## Apply license to EKS Anywhere cluster

> A license can only be bound to one EKS Anywhere cluster at a time, and you can only receive support for your EKS Anywhere cluster if it has a valid and active license. You can only create or update EKS Anywhere clusters with extended support for Kubernetes versions if there is a valid and active license token available for the cluster. Extended support for Kubernetes versions is available in EKS Anywhere versions v0.22 and above.

Reviewer: ```suggestion
A license can only be bound to one EKS Anywhere cluster at a time, and you can only receive support for your EKS Anywhere cluster if it has a valid and active license. You can only create or update EKS Anywhere clusters with extended support for Kubernetes versions if there is a valid and active license token available for the cluster. Extended support for Kubernetes versions is available in EKS Anywhere versions v0.22.0 and above.
```

> #### Apply license to EKS Anywhere cluster with version v0.22 or above

Reviewer: ```suggestion
#### Apply license to EKS Anywhere cluster with version v0.22.0 or above
```

kind: Cluster
metadata:
  name: my-cluster
spec:
  kubernetesVersion: "1.28"
  licenseToken: "eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJsaWNlbnNlSUQiOiJsLTRmMzZhY2YxMmU2ZDQ5MTQ4NDgxMjkyN2IzMjdjMDY2IiwibGljZW5zZVZlcnNpb24iOiIxIiwiYmVnaW5WYWxpZGl0eSI6IjIwMjQtMTAtMTBUMTM6MzM6MzYuMDAwWiIsImVuZFZhbGlkaXR5IjoiMjAyNS0xMC0xMVQxMzozMzozNi4wMDBaIiwic3Vic2NyaXB0aW9uSUQiOiJlMjlmZDBkMi1kOGE4LTRlZDQtYmU1NC1jNmMwZGQwZjc5NjQ6OTQ0NGJkMGQiLCJzdWJzY3JpcHRpb25OYW1lIjoibXktc3Vic2NyaXB0aW9uIiwiYWNjb3VudElEIjoiNjU5MzU2Mzk4NDA0IiwicmVnaW9uIjoidXMtd2VzdC0yIn0.3V1IKL7e0MixnN70KHLZFc1arnzeQRQqF9PuCgUwnOYc_LWHHnvnp-id7i5WUQ7As7DOWKIMMgL6gkTvNhlcMg"
  ...
```
To apply the license token to your cluster, run the `eksctl anywhere create` or `eksctl anywhere upgrade` command, or use Kubernetes API-compatible tooling for workload clusters.

**eksctl anywhere CLI**

New cluster
```
eksctl anywhere create cluster -f my-cluster.yaml --kubeconfig my-cluster.kubeconfig
```
Existing cluster
```
eksctl anywhere upgrade cluster -f my-cluster.yaml --kubeconfig my-cluster.kubeconfig
```
**Kubernetes API-compatible tooling**
```
kubectl apply -f my-cluster.yaml --kubeconfig my-cluster.kubeconfig
```

> #### Apply license to EKS Anywhere cluster with version v0.21 or below

Reviewer: ```suggestion
#### Apply license to EKS Anywhere cluster with version v0.21.x or below
```
